omnigibson/robots/r1.py

```diff
@@ -0,0 +1,519 @@
+        # Set the world-to-base fixed joint to be at the robot's current pose
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localPos0").Set(tuple(position))
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localRot0").Set(
+            lazy.pxr.Gf.Quatf(*orientation[[3, 0, 1, 2]].tolist())
+        )
+
+    # Name of the actual root link that we are interested in. Note that this is different from self.root_link_name,
+    # which is "base_footprint_x", corresponding to the first of the 6 1DoF joints to control the base.
+    @property
+    def base_footprint_link_name(self):
+        return "base_link"
+
+    def _postprocess_control(self, control, control_type):
+        # Run super method first
+        u_vec, u_type_vec = super()._postprocess_control(control=control, control_type=control_type)
+
+        # Change the control from base_footprint_link ("base_footprint") frame to root_link ("base_footprint_x") frame
+        base_orn = self.base_footprint_link.get_orientation()
+        root_link_orn = self.root_link.get_orientation()
+
+        cur_orn = T.mat2quat(T.quat2mat(root_link_orn).T @ T.quat2mat(base_orn))
+
+        # Rotate the linear and angular velocity to the desired frame
+        lin_vel_global, _ = T.pose_transform(th.zeros(3), cur_orn, u_vec[self.base_idx[:3]], th.tensor([0, 0, 0, 1]))
+        ang_vel_global, _ = T.pose_transform(th.zeros(3), cur_orn, u_vec[self.base_idx[3:]], th.tensor([0, 0, 0, 1]))
+
+        u_vec[self.base_control_idx] = th.tensor([lin_vel_global[0], lin_vel_global[1], ang_vel_global[2]])
+        return u_vec, u_type_vec
+
+    def _get_proprioception_dict(self):
+        dic = super()._get_proprioception_dict()
+
+        # Add trunk info
+        joint_positions = ControllableObjectViewAPI.get_joint_positions(self.articulation_root_path)
+        joint_velocities = ControllableObjectViewAPI.get_joint_velocities(self.articulation_root_path)
+        dic["trunk_qpos"] = joint_positions[self.trunk_control_idx]
+        dic["trunk_qvel"] = joint_velocities[self.trunk_control_idx]
+
+        return dic
+
+    @property
+    def control_limits(self):
+        # Overwrite the control limits with the maximum linear and angular velocities for the purpose of clip_control
+        # Note that when clip_control happens, the control is still in the base_footprint_link ("base_footprint") frame
+        # Omniverse still thinks these joints have no limits because when the control is transformed to the root_link
+        # ("base_footprint_x") frame, it can go above this limit.
+        limits = super().control_limits
+        limits["velocity"][0][self.base_idx[:3]] = -m.MAX_LINEAR_VELOCITY
+        limits["velocity"][1][self.base_idx[:3]] = m.MAX_LINEAR_VELOCITY
+        limits["velocity"][0][self.base_idx[3:]] = -m.MAX_ANGULAR_VELOCITY
+        limits["velocity"][1][self.base_idx[3:]] = m.MAX_ANGULAR_VELOCITY
+        return limits
+
+    def get_control_dict(self):
```

Comment:
I think we should remove this...I believe this was causing some control bugs (i.e.: the right arm drifts upwards) and so I had just removed this from Tiago in my recent osc pr